src/components/contextual/pages/pool/invest/InvestPageInvestBlock.vue

```diff
@@ -0,0 +1,77 @@
+<script setup lang="ts">
+import { onBeforeMount } from 'vue';
+import useInvestState from '@/components/forms/pool_actions/InvestForm/composables/useInvestState';
+import InvestForm from '@/components/forms/pool_actions/InvestForm/InvestForm.vue';
+import TradeSettingsPopover, {
+  TradeSettingsContext,
+} from '@/components/popovers/TradeSettingsPopover.vue';
+import usePoolTransfers from '@/composables/contextual/pool-transfers/usePoolTransfers';
+import { usePool } from '@/composables/usePool';
+import { forChange } from '@/lib/utils';
+import { configService } from '@/services/config/config.service';
+import InvestFormV2 from '@/components/forms/pool_actions/InvestForm/InvestFormV2.vue';
+import useInvestPageTabs, { tabs } from '@/composables/pools/useInvestPageTabs';
+
+type Props = {
+  poolSupportsGeneralisedJoin: boolean;
+};
+
+const props = withDefaults(defineProps<Props>(), {});
+
+/**
+ * COMPOSABLES
+ */
+const { network } = configService;
+const { pool, loadingPool, transfersAllowed } = usePoolTransfers();
+const { sor, sorReady } = useInvestState();
+const { activeTab } = useInvestPageTabs();
+const { isDeepPool } = usePool(pool);
+
+/**
+ * CALLBACKS
+ */
+onBeforeMount(async () => {
+  await forChange(loadingPool, false);
+
+  if (props.poolSupportsGeneralisedJoin) {
+    // Initialise SOR for batch swap queries
+    sorReady.value = await sor.fetchPools();
+  } else {
+    sorReady.value = true;
+  }
+});
+</script>
+
+<template>
+  <BalLoadingBlock
+    v-if="loadingPool || !transfersAllowed || !sorReady || !pool"
+    class="h-96"
+  />
+  <BalCard v-else shadow="xl" exposeOverflow noBorder>
+    <template #header>
+      <div class="w-full">
+        <div class="text-xs leading-none text-secondary">
+          {{ network.chainName }}
+        </div>
+        <div class="flex justify-between items-center">
+          <h4>{{ $t('investInPool') }}</h4>
+          <TradeSettingsPopover :context="TradeSettingsContext.invest" />
+        </div>
+        <BalTabs
+          v-if="isDeepPool"
+          v-model="activeTab"
+          :tabs="tabs"
+          class="p-0 m-0 -mb-px whitespace-nowrap"
+          noPad
+        />
+      </div>
+    </template>
+    <template v-if="poolSupportsGeneralisedJoin">
```

Comment:
If we get rid of the prop drilling for this long prop, we can just replace this with isDeep for now.
```suggestion
    <template v-if="isDeepPool">
```

Reply:
True. I removed the `poolSupportsGeneralisedJoin` prop